library/src/main/java/com/google/android/exoplayer/hls/HlsChunkSource.java

```diff
@@ -718,13 +767,52 @@ private boolean shouldRerequestLiveMediaPlaylist(int nextVariantIndex) {
     return timeSinceLastMediaPlaylistLoadMs >= (mediaPlaylist.targetDurationSecs * 1000) / 2;
   }
 
-  private int getLiveStartChunkMediaSequence(int variantIndex) {
+
+  private int getLiveChunkMediaSequence(int variantIndex, long playbackPositionUs, boolean isReset) {
     // For live start playback from the third chunk from the end.
     HlsMediaPlaylist mediaPlaylist = variantPlaylists[variantIndex];
-    int chunkIndex = mediaPlaylist.segments.size() > 3 ? mediaPlaylist.segments.size() - 3 : 0;
+    int chunkIndex;
+    // playbackPositionUs 0 is a special case of seeking/reseting to start playback from
+    // live edge
+    if (playbackPositionUs == 0) {
+      chunkIndex = mediaPlaylist.segments.size() > LIVE_EDGE_CHUNK_INDEX_OFFSET ?
+              mediaPlaylist.segments.size() - LIVE_EDGE_CHUNK_INDEX_OFFSET : 0;
+    } else {
+      availableRange[selectedVariantIndex].getCurrentBoundsUs(availableRangeBoundsUs);
+      long relativePlaybackPositionUs = Math.abs(playbackPositionUs)  - availableRangeBoundsUs[0];
+      if (isReset) {
+        // for seek case we try to find the right chunk index.
+        // if seek position is less than lower bound we cap it at lower bound.
+        // if seek position is greater than upper bound, we cap it at the live edge
+        if (playbackPositionUs > availableRangeBoundsUs[1]) {
```

Comment:
This code can be simplified, I think. If the capping of playbackPositionUs to be within bounds is done before the relativePlaybackPositionUs calculation- at line 782, some of the if/else checks that follows can be avoided. 
Then Util.binarySearchFloor can be used in both cases. 
If the position has reached (beyond) the end, and isReset is false, chunkIndex = mediaPlaylist.segments.size() can be returned, to keep the current method return value behaviour.